component/src/main/java/io/siddhi/extension/io/cdc/source/polling/strategies/DefaultPollingStrategy.java

```diff
@@ -76,52 +107,41 @@ public void poll() {
             statement = connection.prepareStatement(selectQuery);
-
-            while (true) {
-                if (paused) {
-                    pauseLock.lock();
-                    try {
-                        while (paused) {
-                            pauseLockCondition.await();
-                        }
-                    } catch (InterruptedException ie) {
-                        Thread.currentThread().interrupt();
-                    } finally {
-                        pauseLock.unlock();
-                    }
-                }
-                try {
-                    statement.setString(1, lastReadPollingColumnValue);
-                    resultSet = statement.executeQuery();
-                    metadata = resultSet.getMetaData();
-                    while (resultSet.next()) {
-                        detailsMap = new HashMap<>();
-                        for (int i = 1; i <= metadata.getColumnCount(); i++) {
-                            String key = metadata.getColumnName(i);
-                            Object value = resultSet.getObject(key);
-                            detailsMap.put(key.toLowerCase(Locale.ENGLISH), value);
-                        }
-                        lastReadPollingColumnValue = resultSet.getString(pollingColumn);
-                        handleEvent(detailsMap);
-                    }
-                } catch (SQLException ex) {
-                    log.error(buildError("Error occurred while processing records in table %s.", tableName), ex);
-                } finally {
-                    CDCPollingUtil.cleanupConnection(resultSet, null, null);
-                }
-                try {
-                    Thread.sleep((long) pollingInterval * 1000);
-                } catch (InterruptedException e) {
-                    log.error(buildError("Error while polling the table %s.", tableName), e);
+            statement.setString(1, lastReadPollingColumnValue);
+            resultSet = statement.executeQuery();
+            metadata = resultSet.getMetaData();
+            while (resultSet.next()) {
+                detailsMap = new HashMap<>();
+                for (int i = 1; i <= metadata.getColumnCount(); i++) {
+                    String key = metadata.getColumnName(i);
+                    Object value = resultSet.getObject(key);
+                    detailsMap.put(key.toLowerCase(Locale.ENGLISH), value);
                 }
+                lastReadPollingColumnValue = resultSet.getString(pollingColumn);
+                handleEvent(detailsMap);
             }
         } catch (SQLException ex) {
-            throw new CDCPollingModeException(buildError("Error in polling for changes on %s.", tableName), ex);
+            log.error(buildError("Error occurred while processing records in table %s.", tableName), ex);
         } finally {
-            CDCPollingUtil.cleanupConnection(resultSet, statement, connection);
```

Comment:
We should close connection and statement after using. And re-initialize when using again.

Reply:
Connection and statement are closed at the end of the method.